### PR TITLE
[Docs] Adding suggestion sections to high level client docs

### PR DESCRIPTION
**Pull request by cbuescher**
This adds a section about how to add suggestions to the SearchSourceBuilder and
how to retrieve them from a SearchResponse.

### CODE REVIEW

File: docs/java-rest/high-level/apis/search.asciidoc [hunk 1]
```diff
@@ -96,6 +96,23 @@ include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-aggregatio
 
 We will later see how to <<java-rest-high-retrieve-aggs,access aggregations>> in the `SearchResponse`.
 
+===== Requesting Suggestions
+
+To add Suggestions to the search request, use one of the `SuggestionBuilder` implementations
+that are easily accessible from the `SuggestBuilders` factory class. Suggesion builders
+need to be added to the top level `SuggestBuilder` before setting them on the  `SearchSourceBuilder`.
```
Comment (javanna) [inline docs/java-rest/high-level/apis/search.asciidoc]: do they really need to be added "before" ?

**Review by javanna — APPROVED** [review]
left two tiny comments, LGTM otherwise

**Comment by cbuescher** [comment]
@javanna thanks for the review, I changed the two thing you mentioned.

**Comment by javanna** [comment]
thanks @cbuescher go push it then ;)

**Pull request merged**